### PR TITLE
Add a view inspector

### PR DESCRIPTION
**Pull request by Zoxc**
This adds an inspector window for views which lets you view the view tree and layout of views. It only does render captures with the `vger` renderer.

It's hooked up to the widget gallery and counter example (press F11 to open it). It's currently quite slow so I recommend testing it on the counter example.

### CODE REVIEW

**Comment by dzhou121** [comment]
Impressive work!

If I understand it correctly, the capture is to provide a snapshot of the whole UI so that you can give the area when you chose a view in the view tree.

Would it better to have an overlay on the original root view (just like the bit I did for Linux context menu), so that you can directly highlight the area of a view on the original UI, without the need to do a "screenshot", also it's more "real time".

File: src/inspector.rs [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
+        move || selected.get(),
+        move |current| {
+            let info = |i| text(i).style(|s| s.padding(5.0));
+            if let Some(view) = current.and_then(|id| capture.root.find(id)) {
+                let name = info(format!("Type: {}", view.name));
+                let count = info(format!("Child Count: {}", view.children.len()));
+                let x = info(format!("X: {}", view.layout.x0));
+                let y = info(format!("Y: {}", view.layout.y0));
+                let w = info(format!("Width: {}", view.layout.width()));
+                let h = info(format!("Width: {}", view.layout.height()));
```
Comment (presiyan-ivanov) [inline src/inspector.rs:258]:
```suggestion
                let h = info(format!("Height: {}", view.layout.height()));
```

**Comment by Zoxc** [comment]
I intentionally went with an approach with immutable UI snapshots. This keeps the inspector simple and it's helpful for debugging temporal issues. It could be extended to capture multiple frames for animation debugging.

Having an overlay on the real UI with this approach could easily get out of sync so I think that could be confusing or weird. It's also a bit more complex.

**Comment by dzhou121** [comment]
I was thinking of a browser inspector like of a tool.

**Comment by Zoxc** [comment]
I added a `get_content_rect` method to `PaintCx` and used it for `img`. It's used to get the rect inside padding and borders when painting.

**Comment by dzhou121** [comment]
It crashes on my mac due to wrong TextureFormat

```
Incompatible color attachments at indices [0]: the RenderPass uses textures with formats [Some(Rgba8Unorm)] but the RenderPipeline uses attachments with formats [Some(Bgra8Unorm)]
```
So you might need to use the texture_format that's found.

**Comment by dzhou121** [comment]
also window scaling isn't applied by the looks of it. 

Screenshot 2023-11-01 at 21 34 11

**Comment by Zoxc** [comment]
I think I have those issues fixed now.

**Comment by Zoxc** [comment]
I added inspection of view styles too (with colors).

**Pull request merged**